modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java

```diff
@@ -148,4 +182,82 @@ public void close() throws IOException {
         }
     }
 
+    @Override
+    public List<PersistentTasksExecutor<?>> getPersistentTasksExecutor(ClusterService clusterService, ThreadPool threadPool,
+                                                                       Client client, SettingsModule settingsModule,
+                                                                       IndexNameExpressionResolver expressionResolver) {
+        if (GEOIP_V2_FEATURE_FLAG_ENABLED) {
+            Settings settings = settingsModule.getSettings();
+            return List.of(new GeoIpDownloaderTaskExecutor(client, new HttpClient(), clusterService, threadPool, settings));
+        } else {
+            return List.of();
+        }
+    }
+
+    @Override
+    public List<NamedXContentRegistry.Entry> getNamedXContent() {
+        return List.of(new NamedXContentRegistry.Entry(PersistentTaskParams.class, new ParseField(GEOIP_DOWNLOADER),
+                GeoIpTaskParams::fromXContent),
+            new NamedXContentRegistry.Entry(PersistentTaskState.class, new ParseField(GEOIP_DOWNLOADER), GeoIpTaskState::fromXContent));
+    }
+
+    @Override
+    public List<NamedWriteableRegistry.Entry> getNamedWriteables() {
+        return List.of(new NamedWriteableRegistry.Entry(PersistentTaskState.class, GEOIP_DOWNLOADER, GeoIpTaskState::new),
+            new NamedWriteableRegistry.Entry(PersistentTaskParams.class, GEOIP_DOWNLOADER, GeoIpTaskParams::new));
+    }
+
+    @Override
+    public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
+        SystemIndexDescriptor geoipDatabasesIndex = SystemIndexDescriptor.builder()
+            .setIndexPattern(DATABASES_INDEX)
+            .setDescription("GeoIP databases")
+            .setMappings(mappings())
+            .setSettings(Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
```

Comment:
Maybe add `.put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)` here as well?
The default is `1`, which then gets overwritten by auto expand replicas, this is a bit confusing,
so I think setting number of replicas to 0 is clearer, because that is the lower bound of auto expand replicas.

Reply:
Good idea, I've added it